#include <windows.h>\n#include <cppunit/Test.h>\n#include <cppunit/TestResult.h>\n#include <cppunit/TestRunner.h>\n\nclass ExceptionHandler {\npublic:\n    static void handleException(EXCEPTION_POINTERS* pExcept) {\n        // Record the test failure\n        CPPUNIT_TEST_RESULT()->addError(new CppUnit::Exception("Exception caught"));\n    }\n};\n\nint main() {\n    // Set up the exception handler\n    SetUnhandledExceptionFilter((LPTOP_LEVEL_EXCEPTION_FILTER)ExceptionHandler::handleException);\n\n    // Run the tests\n    CppUnit::TestRunner runner;\n    bool success = runner.run(CPPUNIT_TEST_SUITE_REGISTRATION());\n\n    return success ? 0 : 1;\n}